### PR TITLE
Introducing Orchard Core Guru on Gurubase.io

### PR DESCRIPTION
**Pull request by kursataktas**
Hello team,

I'm the maintainer of [Anteon](https://github.com/getanteon/anteon). We have created Gurubase.io with the mission of building a centralized, open-source tool-focused knowledge base. Essentially, each "guru" is equipped with custom knowledge to answer user questions based on collected data related to that tool.

I wanted to update you that I've manually added the [Orchard Core Guru](https://gurubase.io/g/orchard-core) to Gurubase. Orchard Core Guru uses the data from this repo and data from the [docs](https://docs.orchardcore.net/) to answer questions by leveraging the LLM.

In this PR, I showcased the "Orchard Core Guru", which highlights that Orchard Core now has an AI assistant available to help users with their questions. Please let me know your thoughts on this contribution.

Additionally, if you want me to disable Orchard Core Guru in Gurubase, just let me know that's totally fine.

### CODE REVIEW

**Comment by MikeAlhayek** [comment]
@kursataktas Thank you! Please agree to the Contributor License Agreement(CLA)

**Comment by kursataktas** [comment]
@dotnet-policy-service agree

File: README.md
Comment (MikeAlhayek) [inline README.md:14]: I would move this under the documentation on line 12.

**Review by MikeAlhayek — APPROVED** [review]

File: README.md
Comment (kursataktas) [reply README.md]: done 🚀

**Review by Piedone — CHANGES_REQUESTED** [review]
Ah, very cool! A couple of things:

1. Could you link it from [the tutorials page](https://docs.orchardcore.net/en/latest/resources/tutorials/) too? See https://docs.orchardcore.net/en/latest/guides/contributing/contributing-documentation/.
2. How about including these too in the training data?
- https://github.com/OrchardCMS/OrchardCore.Commerce
- https://github.com/OrchardCMS/TryOrchardCore
- https://github.com/OrchardCMS/OrchardCore.Samples
- https://orcharddojo.net/
- https://github.com/Lombiq/Orchard-Training-Demo-Module
3. This is just +1: Can't you allow follow-up questions? Currently, you ask a question, you get an answer, but if that's not perfect, your only option is to try to ask a new, better question.

See this @sfmskywalker.

**Comment by sfmskywalker** [comment]
This sounds super interesting. I’ve been wanting to replace Glime since they shutdown, and this might be a viable alternative. Thanks for the tip!

**Comment by sfmskywalker** [comment]
@kursataktas are there any procedures one might follow to get another OSS project added. Maybe something like https://github.com/elsa-workflows/elsa-core?

**Comment by kursataktas** [comment]
> Ah, very cool! A couple of things:
> 
> 1. Could you link it from [the tutorials page](https://docs.orchardcore.net/en/latest/resources/tutorials/) too? See https://docs.orchardcore.net/en/latest/guides/contributing/contributing-documentation/.
> 2. How about including these too in the training data?
> 
> * https://github.com/OrchardCMS/OrchardCore.Commerce
> * https://github.com/OrchardCMS/TryOrchardCore
> * https://github.com/OrchardCMS/OrchardCore.Samples
> * https://orcharddojo.net/
> * https://github.com/Lombiq/Orchard-Training-Demo-Module
> 
> 3. This is just +1: Can't you allow follow-up questions? Currently, you ask a question, you get an answer, but if that's not perfect, your only option is to try to ask a new, better question.
> 
> See this @sfmskywalker.

Thanks for the review ❤️ 

1. I've updated the PR accordingly. Let me know if you’d like any changes to the text.
2. Sure, I can do that. Gurubase indexes text-based and video files, so documentation and GitHub READMEs are fine. However, we haven’t tested the raw code files, so I don’t recommend including them. I can add the README files from these repositories if you’d like.

Additionally, I found some helpful blog posts on https://orcharddojo.net/ that I can add as a datasource for Orchard Code Guru. However, some of them might be outdated, which could lead to incorrect answers. Is there a time threshold to discard old blog posts, or should I include all of them? Your guidance on this would be helpful.

```
https://orcharddojo.net/blog/4-ways-to-display-something-from-your-module-nested-within-a-page-in-orchard-core-orchard-nuggets
https://orcharddojo.net/blog/add-deployment-for-microsoft-account-deprecate-site-owner-permission-and-retain-administrator-role-as-a-system-role-this-week-in-orchard-01-11-2024
https://orcharddojo.net/blog/add-title-to-background-tasks-and-improve-the-ui-lombiq-node-js-extensions-this-week-in-orchard-17-02-2023
https://orcharddojo.net/blog/advanced-orchard-accessing-other-tenants-services
https://orcharddojo.net/blog/another-lombiq-site-was-improved-orchard-dojo
https://orcharddojo.net/blog/basic-orchard-course-at-obuda-university
https://orcharddojo.net/blog/blazor-guide-for-decoupled-cms-workflow-trimming-task-this-week-in-orchard-06-09-2024
https://orcharddojo.net/blog/content-item-and-widget-alternates-based-on-the-alias-and-slug-url-rewriting-module-this-week-in-orchard-25-10-2024
https://orcharddojo.net/blog/delivering-a-node-js-asset-pipeline-as-a-nuget-package
https://orcharddojo.net/blog/don-t-configure-jsonoptions-by-default-orchard-harvest-2024-videos-are-up-this-week-in-orchard-08-11-2024
https://orcharddojo.net/blog/external-authentication-feature-orchard-nuggets-how-to-add-workflows-to-an-asp-net-core-app-this-week-in-orchard-18-10-2024
https://orcharddojo.net/blog/how-can-i-call-an-external-api-from-a-workflow-task-orchard-nuggets
https://orcharddojo.net/blog/how-to-access-services-from-another-tenant-in-orchard-core-orchard-nuggets
https://orcharddojo.net/blog/how-to-add-a-breadcrumb-menu-in-orchard-core-orchard-nuggets
https://orcharddojo.net/blog/how-to-add-a-culture-url-segment-for-localization-in-orchard-core-orchard-nuggets
https://orcharddojo.net/blog/how-to-add-an-audit-trail-to-an-asp-net-core-app-orchard-core-nuggets
https://orcharddojo.net/blog/how-to-add-workflows-to-an-asp-net-core-app-orchard-core-nuggets
https://orcharddojo.net/blog/how-to-change-the-idle-logout-time-in-orchard-core-orchard-core-nuggets
https://orcharddojo.net/blog/how-to-debug-a-nuget-based-orchard-core-solution-orchard-nuggets
https://orcharddojo.net/blog/how-to-do-a-security-scan-of-an-orchard-core-app-orchard-core-nuggets
https://orcharddojo.net/blog/how-to-fix-invaliddataexception-form-value-count-limit-1024-exceeded-in-orchard-core-orchard-nuggets
https://orcharddojo.net/blog/how-to-localize-content-items-orchard-nuggets
https://orcharddojo.net/blog/how-to-migrate-an-orchard-1-application-to-orchard-core
https://orcharddojo.net/blog/intensive-orchard-core-training-for-the-in-motion-team
https://orcharddojo.net/blog/live-from-orchard-harvest-2024-las-vegas-day-2
https://orcharddojo.net/blog/one-week-intensive-orchard-training-for-the-national-institute-for-clinical-excellence-in-manchester
https://orcharddojo.net/blog/open-dojo-course-and-university-orchard-course
https://orcharddojo.net/blog/orchard-cms-tutorial-recipes-in-orchard-cms
https://orcharddojo.net/blog/orchard-core-2-0-orchard-harvest-2024-this-week-in-orchard-11-10-2024
https://orcharddojo.net/blog/orchard-core-2-0-orchard-harvest-2024-this-week-in-orchard-20-09-2024
https://orcharddojo.net/blog/orchard-core-gallery-orchard-core-2-0-2-this-week-in-orchard-04-10-2024
https://orcharddojo.net/blog/orchard-core-workshops
https://orcharddojo.net/blog/orchard-harvest-2024-montage-video-orchard-dojo-case-study-this-week-in-orchard-27-09-2024
https://orcharddojo.net/blog/our-full-orchard-core-tutorial-series-the-dojo-course-3-is-here
https://orcharddojo.net/blog/training-the-microsoft-mcio-team-on-how-to-develop-orchard
```

3. We are currently developing the follow-up feature, and I believe it will be available in a few weeks. I highly recommend joining our [Discord channel](https://discord.gg/9CMRSQPqx6) and following our [Twitter account](https://x.com/gurubaseio) to stay updated on new features.

**Review by Piedone — APPROVED** [review]

**Comment by Piedone** [comment]
1. Thanks! Changed it a bit, and will merge it once the checks pass.
2. Ah I see, so Gurubase is not processing code. Adding that would be probably the biggest jump, I think, because for OSS projects usually the documentation is not complete. However, the codebase will likely contain all the information you need to work with the project (well, the actual implementation is there, of course, but it'll also have samples and tests that demonstrate usage in an up-to-date fashion).
There's a cutoff, yes, anything before 2020 is likely outdated. From your list, these are not applicable today (for Orchard Core, the latest edition of Orchard):

- https://orcharddojo.net/blog/advanced-orchard-accessing-other-tenants-services but https://orcharddojo.net/blog/how-to-access-services-from-another-tenant-in-orchard-core-orchard-nuggets is the current version
- https://orcharddojo.net/blog/orchard-cms-tutorial-recipes-in-orchard-cms
- https://orcharddojo.net/blog/orchard-cms-tutorial-recipes-in-orchard-cms
3. Great! I've joined your Discord.

**Comment by Piedone** [comment]
@all-contributors please add @kursataktas for documentation.

**Comment by kursataktas** [comment]
>    There's a cutoff, yes, anything before 2020 is likely outdated. From your list, these are not applicable today (for Orchard Core, the latest edition of Orchard):
> 
> * https://orcharddojo.net/blog/advanced-orchard-accessing-other-tenants-services but https://orcharddojo.net/blog/how-to-access-services-from-another-tenant-in-orchard-core-orchard-nuggets is the current version
> * https://orcharddojo.net/blog/orchard-cms-tutorial-recipes-in-orchard-cms
> * https://orcharddojo.net/blog/orchard-cms-tutorial-recipes-in-orchard-cms

I added these blogs as data source for Orchard Guru by eliminating the old ones according to your comment. 🎊 

I also responded to the other topics on Discord, fyi.

**Comment by Piedone** [comment]
Great, thank you!

**Comment by kursataktas** [comment]
Hi everyone,

I’d like to update you on the release of the Maintainer Panel feature on Gurubase. With this panel, you can add, remove, or update data sources, as well as change the logo and more. You can find the details [here](https://github.com/getanteon/gurubase?tab=readme-ov-file#how-to-claim-a-guru). 

In the near future, I’m planning to include analytics insights in this panel, such as the number of questions asked, the most frequently asked ones, and more. I’ll be sure to update this thread once it’s available. However, in case I miss it, I highly recommend joining our [Discord channel](https://discord.gg/9CMRSQPqx6) to stay updated.

If you’d prefer that I don’t update this thread anymore, please let me know.

**Comment by Piedone** [comment]
Does that mean we can add the codebase as a source too?

Having updates under this PR is nice, though I don't think many people apart from those who commented here will see it. For that, https://github.com/OrchardCMS/OrchardCore/discussions is better.

**Comment by kursataktas** [comment]
> Does that mean we can add the codebase as a source too?
No, this is not available yet. In the initial version, you can only manage data sources in the formats of webpages, YouTube videos, or PDFs.

**Comment by Piedone** [comment]
I see, OK then.

**Comment by kursataktas** [comment]
Hello Everyone,

We recently released the embeddable "Ask AI" widget, which allows you to integrate your Guru’s Q&A capability into websites, documentation, and more.

The installation process is straightforward, as explained in [this GitHub repository](https://github.com/getanteon/gurubase-widget).

You can see it in action on the [Anteon](https://getanteon.com/) and [NetAlertX](https://netalertx.com/) websites.

**Comment by kursataktas** [comment]
Hello Everyone,

I want to share an exciting update about Gurubase. We've recently released several great features that I think you'll find interesting.

### 1. Slack & Discord Integration
You can now add your Guru as a bot to Slack and Discord, allowing you to ask questions instantly. More details here: [Discord Bot Integration](https://docs.gurubase.ai/integrations/discord-bot).

### 2. Analytics Dashboard
Track what your users ask your Guru and gain valuable insights. This feature is available on the "My Gurus" page. If you haven’t maintained your Guru yet, I highly recommend doing so to access your analytics. More information about claiming your Guru can be found [here](https://github.com/Gurubase/gurubase?tab=readme-ov-file#how-to-claim-a-guru).
image

### 3. API Support
All Guru functionalities are now accessible via API. You can ask questions, edit data sources, and more using our API endpoints. Here is the [API Doc](https://docs.gurubase.ai/api-reference/introduction).

### 4. Codebase Indexing
We now support indexing codebases from GitHub repositories to generate answers for related queries. This feature is available for select Gurus. If you don’t see the "Codebase Indexed" label on your Guru edit page and would like access, please let me know, I’d be happy to enable it for you!
image

Let me know if you have any questions or feedback!

**Comment by Piedone** [comment]
Great! I can't edit https://gurubase.io/g/orchard-core so I can't check, but is that indexing the codebase? If not, could you please enable?

**Comment by kursataktas** [comment]
@Piedone I've assigned it to your Gurubase account. You can find it on the 'My Gurus' page (click the profile icon, and you should see the "My Gurus" tab). I’ve enabled codebase indexing for Orchard Core, but the size of the repo is hitting the hard limits. I'll make some quick adjustments to manually specify limits for important repositories, as Orchard Core Guru is one of the most used Gurus on the platform. In the meantime, I highly suggest you take a look at the Analyze tab, it will likely provide great insights into what your users are asking about Orchard Core and what information is lacking in your docs. I'll update this thread once the codebase has been indexed.

**Comment by kursataktas** [comment]
> Great, thanks, I can see it. Can multiple people from our community have access to it?

Yes, you can send me the Gurubase username/email addresses for them, and I'll handle the rest.

I have enabled the codebase indexing for Orchard Core. Gurubase reindexes the repo every 12 hours to stay updated. Please let me know if you have any issues.

**Comment by Piedone** [comment]
Oh great! I've told this the maintainers, they'll let you know. Two more things:

1. As I mentioned previously too, it would also great to index https://github.com/OrchardCMS/OrchardCore.Commerce, https://github.com/OrchardCMS/TryOrchardCore, https://github.com/OrchardCMS/OrchardCore.Samples, and https://github.com/Lombiq/Orchard-Training-Demo-Module. I don't see a way to add code repos under https://gurubase.io/guru/orchard-core though. Can you index them?
2. Can one add a YouTube channel to be indexed as whole, or at least a playlist? Because adding https://www.youtube.com/@Lombiq or https://www.youtube.com/playlist?list=PLuskKJW0FhJfOAN3dL0Y0KBMdG1pKESVn doesn't work with "Error extracting content from the YouTube URL" (I'd also add https://www.youtube.com/playlist?list=PLuskKJW0FhJcF-YxKIRj4IyGVjJEGEkSZ, https://www.youtube.com/playlist?list=PLuskKJW0FhJf_uqOFbozmquvoKeSgKQG2, and https://www.youtube.com/@OrchardCore).

**Comment by kursataktas** [comment]
@Piedone Gurubase doesn't support both of these. But I have created [this](https://github.com/Gurubase/gurubase/issues/113) and [this](https://github.com/Gurubase/gurubase/issues/114) issue to add them on the roadmap. I'll update this PR about the progess.

**Comment by Piedone** [comment]
Thank you, subscribed to those issues.

**Comment by Piedone** [comment]
Thanks for the update! For us, it would be more useful to have the bot watch the discussions; we consciously keep questions out of issues.

**Comment by Piedone** [comment]
I'll follow up under https://github.com/Gurubase/gurubase/issues/112.

**Pull request merged**
